File: APM/Java/Interpreter/src/ro/sandorrobertk94/domain/MyList.java
package ro.sandorrobertk94.domain;

import java.util.Arrays;

/**
 * Created by robert on 10/25/15.
 */
public class MyList implements MyIList {
    private static final int CAPACITY = 50;
    private int size;
    private Object[] objList;

    @Override
    public void add() {
        size = 0;
        objList = new Object[CAPACITY];
    }

    @Override
    public Object get(int pos) {
        if (pos < size) {
            return objList[pos];
        }
        return null;
    }

    @Override
    public int getSize() {
        return size;
    }

    @Override
    public String toString() {
        String out = "[ size =" + size + "\n";
        for (int i = 0; i < size; i++) {
            out += objList[i].toString() + "\n";
        }
        out += "];";
        return out;
    }
}

File: APM/CSharp/Interpreter/Interpreter/domain/MyList.cs
namespace Interpreter.domain
{
    public class MyList : MyIList
    {
        private const int CAPACITY = 50;
        private readonly object[] objList;
        private int size;

        public MyList()
        {
            size = 0;
            objList = new object[CAPACITY];
        }

        public void Add(object obj)
        {
            objList[size] = obj;
            size++;
        }

        public object Get(int pos)
        {
            if (pos < size)
            {
                return objList[pos];
            }
            return null;
        }

        public int getSize()
        {
            return size;
        }

        public override string ToString()
        {
            var o = "[ size = " + size + "\n";
            for (var i = 0; i < size; i++)
            {
                o += "\t" + objList[i] + "\n";
            }
            o += "];";
            return o;
        }
    }
}